### ISSUE TITLE
Problems with payment information returns

### ISSUE THREAD

**Issue by Nutripure** [PrestaShop/PrestaShop#12618]
Hello
we have updated to 1.7.5.0, we are facing problems with validated payment information returns, for paypal and credit cards paiements (systempay Caisse Epargne). Prestashop does not retrieve the validated payment information. This information is clearly visible in the server log. After several attempts, the information goes into fatal error and the order doesn't appear on the back office.

The problem occurs on approximately 15% of the order

Any ideas

**Additionnal information**
PrestaShop version: 1.7.5.0
PHP version: 7.1.25

**Comment by khouloudbelguith** [comment]
Hi @Nutripure,

We need to retrieve the PHP error log and the debug mode report in order to find out what's wrong.
Don't you know how to get this information? Please read the following article:
http://build.prestashop.com/howtos/misc/how-to-create-bug-report/

Thanks!

**Comment by Nutripure** [comment]
Logs in BO are : 

PaymentModule::validateOrder - Secure key does not match


PaymentModule.php -> ligne 253

if ($secure_key !== false && $secure_key != 
$this->context->cart->secure_key) {
                 PrestaShopLogger::addLog('PaymentModule::validateOrder - 
Secure key does not match', 3, null, 'Cart', (int) $id_cart, true);
                 die(Tools::displayError());

This mistake is generated in this file (line 253)

**Comment by khouloudbelguith** [comment]
Hi @Nutripure,

Could you please check if the cart secure key is different than customer secure key when this issue occurs.
Did you delete carts from the Database?
Could you please try to run those command lines to remove the cache:
`rm -fR /Projet_PrestaShop/var/cache/*`

```
php -d memory_limit=-1 ./bin/console cache:clear
php -d memory_limit=-1 ./bin/console cache:clear --env=prod
```
You need also to force refresh the page, by pressing CTRL+F5 on Windows / Linux, Cmd+R on Mac OS while you are on this page.
Thanks to check & feedback.

**Comment by Nutripure** [comment]
Hello

1. we set up an email to check the secure key. On validated orders, key are the same. On bugged orders, we will verify that keys are differents. this problem only occurs 10% of the orders.

2. we never deleted carts from the database

3. we deleted the cache from Projet_PrestaShop/var/cache/prod (unique folder in cache). what is it suppose to produce ? 

thanks!

**Comment by khouloudbelguith** [comment]
@Nutripure, to be sure is not a cache issue, we need to retrieve cache folder manually.
THis issue occurs only with this module "systempay Caisse Epargne", have you tried to contact the module author about this issue?
Thanks!

**Comment by Nutripure** [comment]
Hello
cache folder was removed manually. 
this problem occurs for systempay and paypal. So no need to contact the module authors.
The problems occurs in prestashop file PaymentModule
thanks

**Comment by khouloudbelguith** [comment]
@Nutripure, thanks for your feedback.
I tried with PS1.7.5.0 & the PayPal module v4.4.2 & it is OK.
@marionf what do you think?
Can you reproduce the issue?
Thanks!

**Comment by Nutripure** [comment]
Hello
bug is randomized, it doesn't appear systematically.
After deleting the cache, the errors occurs twice that night.
thanks
Florent

**Comment by marionf** [comment]
@khouloudbelguith I tried also with paypal and can't reproduce this issue

**Comment by marionf** [comment]
Hello,

Despite our several trials, we could not reproduce your issue with the provided information.
It seems that your issue is not a PrestaShop's core bug but most likely a server configuration or customization problem.
I invite you to contact our support team: http://addons.prestashop.com/en/388-support to help you solve your issue.

Best regards.

**Comment by Nutripure** [comment]
After investigating, I found that there is a conflict between the cart number and the order number sent to the paiement module with are not equal when it does bad. 

Normally, cart number is set as the order number in the payment module. When the payment is ok, this number is checked in prestashop BO :
 - if cart number and payement number are the same (the two secury keys are also the same), order is validated and payment number is transformed in BO order in prestashop BO
- if cart number and payment number are not the same (the two security keys are not the same), order is not validated. 
In this last case, when looking into the payment information, I found that payement is attributed to the owner of a wrong cart : it is attributed to the cart which number is corresponding with the payement number (an old abandonned cart).
It make me think that neither two cart have the same number or there are two cart registers that are not matching.

I have to say that the problems occurs much more frequently since last thursday, when I upgraded from 17.2.4 to 1.7.5

thanks for your help
Florent
